### ISSUE TITLE
Fix duplicate row issue

### ISSUE THREAD

**Issue by paf31** [purescript/purescript-exceptions#5]

**Comment by hdgarrood** [comment]
Am I correct in assuming the duplicate row issue is that the following gives you a type error but ought to work?

```
do  catchException (\err -> ...) something
    somethingElseWhichCanThrowExceptions
```

We could perhaps solve this by exporting a separate version of `catchException` for when you don't want to remove the `EXCEPTION` effect?

**Comment by paf31** [comment]
Yes, at one point this extra function used to exist. I'm not sure when it got removed. I think I'd like to add back support for duplicate rows in the longer term.

**Comment by natefaubion** [comment]
Could we add this back in for upcoming new release?

**Comment by paf31** [comment]
I'm thinking of allowing duplicate labels again in the compiler, soon, at which point, this won't be an issue any more. I'd be fine with this as a temporary workaround, but given there is another workaround (`unsafeInterleaveEff`), it is essential, given it'll be a major break to remove it again?

**Comment by natefaubion** [comment]
My main concern is that `catchException`, as-is, is basically unusable except at the very top-level. I haven't encountered a situation where it has not caused issues.

**Comment by garyb** [comment]
Fixed by PS 0.11?

**Comment by paf31** [comment]
I think so.
